### ISSUE TITLE
Discrepancies vs lightning feedstock  (keep both?)

### ISSUE THREAD

**Issue by DManowitz** [conda-forge/pytorch-lightning-feedstock#134]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Both this feedstock and the one for the `lightning` package both reference the same project.  It is concerning that there are 2 packages in conda-forge currently tracking the same project.  Even more concerning is that the dependencies between these 2 packages do not match!  If both conda-forge packages are needed, then they should at least have the same dependencies!

### Installed packages

```shell
N/A
```

### Environment info

```shell
N/A
```

**Comment by Borda** [comment]
True it is tracking one github repository, but within the repo we have multiple packages with different codebase and requirements.

**Comment by DManowitz** [comment]
Would it be possible to add lightning versions prior to 2.0 to conda-forge?  A number of packages are not yet compatible with lightning/pytorch-lightning >=2.0, so it would be good to have some lightning options before v2 on conda-forge to work with those other packages for now.

**Comment by Borda** [comment]
> Would it be possible to add lightning versions prior to 2.0 to conda-forge? A number of packages are not yet compatible with lightning/pytorch-lightning >=2.0, so it would be good to have some lightning options before v2 on conda-forge to work with those other packages for now.

Do you have some specific versions in mind?

**Comment by DManowitz** [comment]
Actually, can you make that 1.9.4 instead of 1.9.5?

**Comment by Borda** [comment]
> Actually, can you make that 1.9.4 instead of 1.9.5?

Sounds good, cound you send a PR?
Cc: @jan-janssen

**Comment by DManowitz** [comment]
> > Actually, can you make that 1.9.4 instead of 1.9.5?
> 
> Sounds good, cound you send a PR? Cc: @jan-janssen

Unfortunately, I don't have time for that at the moment.

**Comment by DManowitz** [comment]
I finally have some time.  Do I just change the `recipe/meta.yaml` to reflect the lightning package, rather than pytorch-lightning, and do I just do this as normal PRs, even though they are for older versions?  cc: @jan-janssen

**Comment by weiji14** [comment]
Ok, I see a PR for a pre-2.0 version of lightning at https://github.com/conda-forge/lightning-feedstock/pull/12, which seems to be adapted from https://github.com/conda-forge/pytorch-lightning-feedstock/pull/140

> A number of packages are not yet compatible with lightning/pytorch-lightning >=2.0, so it would be good to have some lightning options before v2 on conda-forge to work with those other packages for now.

@DManowitz, could you provide a list of those packages that don't work with lightning 2.0? I just want to double check that they actually require `lightning` and not `pytorch-lightning`, because there *is* a pytorch-lightning=1.8.6 already (https://github.com/conda-forge/pytorch-lightning-feedstock/pull/121), and I wonder if you could just depend on that instead of building a lightning=1.8.6 conda-forge package.

**Comment by DManowitz** [comment]
> Ok, I see a PR for a pre-2.0 version of lightning at [conda-forge/lightning-feedstock#12](https://github.com/conda-forge/lightning-feedstock/pull/12), which seems to be adapted from #140
> 
> > A number of packages are not yet compatible with lightning/pytorch-lightning >=2.0, so it would be good to have some lightning options before v2 on conda-forge to work with those other packages for now.
> 
> @DManowitz, could you provide a list of those packages that don't work with lightning 2.0? I just want to double check that they actually require `lightning` and not `pytorch-lightning`, because there _is_ a pytorch-lightning=1.8.6 already (#121), and I wonder if you could just depend on that instead of building a lightning=1.8.6 conda-forge package.

I guess the question is how closely do the `lightning` and `pytorch-lightning` versions have to be to each other.  For example, can `lightning v2.x` be installed together with `pytorch-lightning v1.x`?  If so, then I guess I don't need to have the `lightning v1.x` packages be on conda-forge.

**Comment by Borda** [comment]
They started to be released in sync sync `1.8.0`

**Comment by DManowitz** [comment]
> They started to be released in sync sync `1.8.0`

Do they need to be kept in sync?  In the dependencies for `lightning`, I see only `pytorch-lightning`, not a specific version of `pytorch-lightning`.

**Comment by weiji14** [comment]
> I guess the question is how closely do the `lightning` and `pytorch-lightning` versions have to be to each other. For example, can `lightning v2.x` be installed together with `pytorch-lightning v1.x`? If so, then I guess I don't need to have the `lightning v1.x` packages be on conda-forge.

Let me rephrase this. Do you need to install `lightning` which according to https://github.com/Lightning-AI/lightning/discussions/16688#discussioncomment-6751114, consists of PyTorch Lightning + Lightning Fabric + Lightning Apps + Torchmetrics + Lightning Cloud, etc (i.e, the Lightning Ecosystem), or do you just need `pytorch-lightning` which is the core neural network trainer stuff.

I'd actually be surprised if you have a dependency that depends on lightning v1.x because from experience, the Lightning Apps stuff was pretty unstable in the v1.x version, and looking through several recipe/meta.yaml files for feedstocks at https://github.com/search?q=org%3Aconda-forge+lightning+language%3AYAML&type=code&p=1, I don't think there are any recipes depending on lightning<2.0? But if you do need to use lightning v1.8.6 to be compatible with pytorch-lightning v1.8.6 for whatever reason, we can help out at https://github.com/conda-forge/lightning-feedstock/pull/12.

**Comment by DManowitz** [comment]
> > I guess the question is how closely do the `lightning` and `pytorch-lightning` versions have to be to each other. For example, can `lightning v2.x` be installed together with `pytorch-lightning v1.x`? If so, then I guess I don't need to have the `lightning v1.x` packages be on conda-forge.
> 
> Let me rephrase this. Do you need to install `lightning` which according to [Lightning-AI/lightning#16688 (reply in thread)](https://github.com/Lightning-AI/lightning/discussions/16688#discussioncomment-6751114), consists of PyTorch Lightning + Lightning Fabric + Lightning Apps + Torchmetrics + Lightning Cloud, etc (i.e, the Lightning Ecosystem), or do you just need `pytorch-lightning` which is the core neural network trainer stuff.
> 
> I'd actually be surprised if you have a dependency that depends on lightning v1.x because from experience, the Lightning Apps stuff was pretty unstable in the v1.x version, and looking through several recipe/meta.yaml files for feedstocks at https://github.com/search?q=org%3Aconda-forge+lightning+language%3AYAML&type=code&p=1, I don't think there are any recipes depending on lightning<2.0? But if you do need to use lightning v1.8.6 to be compatible with pytorch-lightning v1.8.6 for whatever reason, we can help out at [conda-forge/lightning-feedstock#12](https://github.com/conda-forge/lightning-feedstock/pull/12).

You are correct in that the packages I'm thinking of (lightning-bolts and lightning-flash) need `pytorch-lightning` v1.x.  However, the question that I don't know the answer to and that I'm hoping one of you can answer is whether I can install `lightning` v2.x together with `pytorch-lightning` v1.x.  If so, then I don't need to have `lightning` v1.x added to conda-forge.

**Comment by lantiga** [comment]
In general I would recommend we keep:
- `lightning` >= 2.x
- `pytorch-lightning` 1.x and 2.x

> If so, then I don't need to have lightning v1.x added to conda-forge.
I agree
